packages/connect/src/protocol-connect/get-request.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Message, protoBase64 } from "@bufbuild/protobuf";
+import type { AnyMessage } from "@bufbuild/protobuf";
+import type { UnaryRequest } from "../index.js";
+import {
+  headerContentType,
+  headerProtocolVersion,
+  headerUnaryAcceptEncoding,
+  headerUnaryContentLength,
+  headerUnaryEncoding,
+} from "./headers.js";
+
+const contentTypePrefix = "application/";
+
+function encodeMessageForUrl(message: Uint8Array, useBase64: boolean): string {
+  if (useBase64) {
+    // TODO(jchadwick-buf): Three regex replaces seems excessive.
+    // Can we make protoBase64.enc more flexible?
```

Comment:
protobuf-es uses this for JSON, and it doesn't seem ideal to add complexity, bundle size there for a feature that is not used by protobuf-es. I agree this isn't pretty, but I am not sure we have a really good alternative.